base/show.jl

```diff
@@ -3128,6 +3128,24 @@ function array_summary(io::IO, a, inds)
     print(io, " with indices ", inds2string(inds))
 end
 
+"""
+    summary(io::IO, ref::GenericMemoryRef)
+
+Print a brief description of a `GenericMemoryRef`, showing the remaining
+accessible length from the current offset. This improves error messages
+like `BoundsError` to reflect the effective size, similar to `SubString`.
+"""
```

Comment:
There's no need to add a docstring for methods that are already documented by the generic function docstring and don't deviate from the generic interface.

```suggestion
```